src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -17,4 +17,14 @@ public abstract class Command {
      */
     public abstract CommandResult execute(Model model) throws CommandException;
 
+    /**
+     * Checks beforehand if the input is valid.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @throws CommandException If an error occurs during command execution.
+     */
+    public void validateInput(Model model) throws CommandException {
+        // Do nothing
```

Comment:
Is it better if this becomes an abstract class while delete and clear class implements it, while other classes do nothing

Reply:
My idea was that if this were abstract, we would have to override it every time we create a new command, whereas if this were a concrete implementation, we only have to override those that needed a specific behavior.

Reply:
oh wait i forgot if you call validateinput in the other classes it will call this. my bad